vulnerabilities/pipelines/v2_importers/istio_importer.py

```diff
@@ -0,0 +1,170 @@
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import ReferenceV2
+from vulnerabilities.pipelines import VulnerableCodeBaseImporterPipelineV2
+from vulnerabilities.utils import get_advisory_url
+from vulnerabilities.utils import split_markdown_front_matter
+
+is_release = re.compile(r"^[\d.]+$", re.IGNORECASE).match
+
+
+class IstioImporterPipeline(VulnerableCodeBaseImporterPipelineV2):
+    """
+    Importer for Istio.io security advisories.
+    """
+
+    pipeline_id = "istio_importer_v2"
+    spdx_license_expression = "Apache-2.0"
+    license_url = "https://github.com/istio/istio.io/blob/master/LICENSE"
+    repo_url = "git+https://github.com/istio/istio.io"
+    unfurl_version_ranges = True
+
+    @classmethod
+    def steps(cls):
+        return (
+            cls.clone,
+            cls.collect_and_store_advisories,
+            cls.clean_downloads,
+        )
+
+    def advisories_count(self) -> int:
+        base_path = Path(self.vcs_response.dest_dir)
+        advisories_dir = base_path / "content/en/news/security"
+        return sum(
+            1 for file in advisories_dir.rglob("*.md") if not file.name.endswith("_index.md")
+        )
+
+    def clone(self):
+        self.log(f"Cloning `{self.repo_url}`")
+        self.vcs_response = fetch_via_vcs(self.repo_url)
+
+    def collect_advisories(self) -> Iterable[AdvisoryData]:
+        base_path = Path(self.vcs_response.dest_dir)
+        advisories_dir = base_path / "content/en/news/security"
+
+        for md_file in advisories_dir.rglob("*.md"):
+            if md_file.name.endswith("_index.md"):
+                continue
+
+            data = self.parse_markdown(md_file)
+            advisory_url = get_advisory_url(
+                file=md_file,
+                base_path=base_path,
+                url="https://github.com/istio/istio.io/blob/master/",
+            )
+            published_date = data.get("publishdate")
+            release_date = (
+                parser.parse(published_date).replace(tzinfo=pytz.UTC) if published_date else None
+            )
+            constraints = self.get_version_constraints(data.get("releases", []))
+
+            cves = data.get("cves", [])
+
+            affected_packages = []
+            if constraints:
+                affected_packages.extend(
+                    [
+                        AffectedPackage(
+                            package=PackageURL(type="golang", namespace="istio.io", name="istio"),
+                            affected_version_range=GolangVersionRange(constraints=constraints),
+                        ),
+                        AffectedPackage(
+                            package=PackageURL(type="github", namespace="istio", name="istio"),
+                            affected_version_range=GitHubVersionRange(constraints=constraints),
+                        ),
+                    ]
+                )
+
+            title = data.get("title") or ""
```

Comment:
Since we are using it as an `advisory_id`, if there is no `title` we should log error and continue, instead for returning a AdvisoryData with empty `advisory_id`.